lib/trento/activity_log.ex

```diff
@@ -42,14 +40,51 @@ defmodule Trento.ActivityLog do
     end
   end
 
-  @spec list_activity_log() :: list(ActivityLog.t())
-  def list_activity_log do
-    # This will be made filterable/paginatable in a later PR
-    query =
-      from activity in ActivityLog,
-        order_by: [desc: activity.inserted_at]
+  @spec list_activity_log(map()) ::
+          {:ok, list(ActivityLog.t()), Flop.Meta.t()} | {:error, :activity_log_fetch_error}
+  def list_activity_log(params) do
+    parsed_params =
+      case params do
+        [] -> %{}
+        params -> parse_params(params)
+      end
+
+    case Flop.validate_and_run(ActivityLog, parsed_params, for: ActivityLog) do
+      {:ok, {activity_log_entries, meta}} ->
+        {:ok, activity_log_entries, meta}
+
+      error ->
+        Logger.error("Activity log fetch error: #{inspect(error)}")
+        {:error, :activity_log_fetch_error}
+    end
+  end
+
+  # ''&& false' is a workaround until we reach OTP 27 that allows doc tag for private functions;
+  # we get a compile warning without this with OTP 26
+  @doc """
+       Parses the query parameters and returns a map with the parsed values as expected by the Flop library.
+       Some parameters are recognized by Flop and are used as is (example: last, first, after, before);
+       some other parameters are used to build filters with custom operator logic (example: from_date, to_date, actor, type).
+       ## Examples
+             iex> parse_params([{:from_date, "2021-01-01"}, {:to_date, "2021-01-31"}, last: 10])
+             %{filters: %{from_date: "2021-01-01", to_date: "2021-01-31"}, last: 10}
```

Comment:
Maybe a typo
```suggestion
             %{filters: %{from_date: "2021-01-01", to_date: "2021-01-31"}, last: 10}
             %{filters: %{inserted_at: %{value: "2021-01-31", op: :>=, field: :inserted_at}}, last: 10}
```

Reply:
@gagandeepb @balanza now that I am looking at this again I noticed that we might be missing something.

if we have both `from_date` and `to_date` as an input, we should expect `inserted_at` to be between those dates

```elixir
# input
# [{:from_date, "2021-01-01"}, {:to_date, "2021-01-31"}]
[
  %{
    filters: [
      %{inserted_at: %{value: "2021-01-01", op: :>=, field: :inserted_at}},
      %{inserted_at: %{value: "2021-01-31", op: :<=, field: :inserted_at}}
    ]
  }
]
```

I guess in the reduction we are not considering multiple filter operations on the same field.

Reply:
@nelsonkopliku so you expect an output like this?

```elixir
[
  %{
    filters: [
      %{inserted_at: %{value: ["2021-01-01", "2021-01-31"], op: :between field: :inserted_at}},
    ]
  }
]
```
(I didn't check Flop docs for the exact implementation of the operator)

Reply:
I did not check Flop's doc either :see_no_evil: but yes: if we provide `from_date` and `to_date` I would expect `inserted_at` to be between those dates.

What we currently generate is a different filter, lacking part of the intent
```elixir
%{
  filters: %{inserted_at: %{value: "2021-01-31", op: :>=, field: :inserted_at}}
}
```

Reply:
I see, there is a bug.

Basically because the `filters` map's key is the field name. Good catch.

Let's check Flop docs for how it expects the filter map to be.

Reply:
Traced the bug in https://github.com/trento-project/web/pull/2786#discussion_r1686329508

We may have some concerns about tests (they should have spoiled the bug)

Reply:
I'll add some more tests.